src/coreclr/tools/ILVerification/ILImporter.Verify.cs

```diff
@@ -2153,7 +2152,7 @@ void ImportStoreField(int token, bool isStatic)
 
                 instance = actualThis.Type;
 
-                if (field.IsInitOnly)
+                if (field.IsInitOnly && !(field.OwningType == _method.OwningType && _method.IsSpecialName && (_method.IsConstructor || _method.Signature.IsExternalInit)))
                     Check(_method.IsConstructor && field.OwningType == _method.OwningType && actualThis.IsThisPtr, VerifierError.InitOnly);
```

Comment:
Should this rather augment the condition in the `Check` method argument?

Reply:
Sure, currently we have mixed style but if `Check` condition is preferred `field.IsInitOnly` can also be inlined.